datafusion/src/dataframe.rs

```diff
@@ -223,6 +223,21 @@ pub trait DataFrame: Send + Sync {
     /// ```
     async fn collect(&self) -> Result<Vec<RecordBatch>>;
 
+    /// Print results.
+    ///
+    /// ```
+    /// # use datafusion::prelude::*;
+    /// # use datafusion::error::Result;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new())?;
+    /// df.show().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn show(&self) -> Result<()>;
```

Comment:
It would be nice if we could have similar behavior to Spark, where the default for `show` is to just show the first 20 rows.

```scala
def show(numRows: Int): Unit = show(numRows, truncate = true)

def show(): Unit = show(20)
```

Reply:
DataFusion df has limit function to limit the number of rows, so I think it is just to add the default 20 rows, what do you think?

Reply:
I think adding a call to `df.limit(20)` prior to showing the output would work well

Reply:
I think it can be inferred from the logical plan whether there is a limit number. If there is no limit, set it to the default value of 20. If there is, then the default limit value is not set.

Reply:
> This is an implementation:

I agree that looks perfect

Reply:
OK, i pushed it, please review this changes.

Reply:
Sorry to be a pain but this seems confusing to me and maybe it would be better to revert to your original code and document that the user can limit output by using `df.limit(20).show()`. Alternatively, we could add a `show_limit(limit: usize)` alternate method where the user can specify how many rows they would like.

The issue I have with this is that it shows 20 rows by default and if the user wants more then they need to add a limit, which is counterintuitive because limit normally reduces the number of rows. Also, this code only works if the final operator is a limit, so it won't work constantly if the limit is wrapped in a sort, for example.

Reply:
I greet, I will rewrite.

Reply:
I defer to @andygrove